lightning/src/ln/channelmonitor.rs

```diff
@@ -1021,7 +1013,7 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 			our_to_self_delay: our_to_self_delay,
 			their_to_self_delay: None,
 
-			old_secrets: [([0; 32], 1 << 48); 49],
+			commitment_secrets: CounterpartyCommitmentSecrets::new(),
```

Comment:
Sounds good!

Reply:
No need to do anything now but would revisit when work on watchtowers. In case of leak of commitment_secret + revocation_basepoint_secret, a misbehaving remote peer would be able to alleviate its punishment so an implementor should be able to get it out-of-memory. Not necessarily behind KeyInterface.